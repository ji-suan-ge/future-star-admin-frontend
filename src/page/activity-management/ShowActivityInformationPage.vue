<template>
  <el-container>
    <el-main>
      <div>
        <el-page-header @back="goBack()" content="活动详情"/>
        <el-divider></el-divider>
      </div>
      <el-card
        class="info-card"
        shadow="never">
        <el-row>
          <el-col
            :span="24"
            class="center-position">
            <div class="block">
              <el-avatar
                :size="130"
                :src="currentActivity.image">
              </el-avatar>
            </div>
          </el-col>
        </el-row>
        <el-form class="info-table" label-width="130px">
          <div class="line">
            <el-divider></el-divider>
          </div>
          <div class="description">- 活动信息 -</div>
          <el-row class="top-distance">
            <el-col :span="width" :offset="2">
              <i class="el-icon-user">&nbsp;活动名称：</i>
              <label>{{ currentActivity.name }}</label>
            </el-col>
            <el-col :span="width">
              <i class="el-icon-male">&nbsp;报名开始时间：</i>
              <label>{{ currentActivity.enroll_start_time }}</label>
            </el-col>
            <el-col :span="width">
              <i class="el-icon-timer">&nbsp;报名结束时间：</i>
              <label>{{ currentActivity.enroll_end_time }}</label>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="width" :offset="offset">
              <i class="el-icon-mobile-phone">&nbsp;活动时间：</i>
              <label>{{ currentActivity.start_time }}</label>
            </el-col>
            <el-col :span="width">
              <i class="el-icon-document">&nbsp;面向人群：</i>
              <label v-for="item in clazzOptions"
                     :key="item.clazz"
                     :label="item.label"
                     :value="item.clazz">
              </label>
            </el-col>
            <el-col :span="width">
              <i class="el-icon-school">&nbsp;活动地点：</i>
              <label >{{ currentActivity.address }}</label>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="width" :offset="offset">
              <i class="el-icon-location-information">&nbsp;限制人数：</i>
              <label>{{ currentActivity.people_number_limit }}</label>
            </el-col>
            <el-col :span="width">
              <i class="el-icon-chat-dot-round">&nbsp;收费情况：</i>
              <label>{{ currentActivity.price }}</label>
            </el-col>
            <el-col :span="width">
              <i class="el-icon-edit-outline">&nbsp;主办方：</i>
              <label >{{ currentActivity.organizer }}</label>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col
              :span="21"
              :offset="offset">
              <i class="el-icon-reading">&nbsp;活动日程：</i>
              <label>{{ currentActivity.arrangement }}</label>
            </el-col>
          </el-row>
        </el-form>
        <el-row>
          <el-col
            :span="24"
            class="center-position">
            <el-button
              type="primary"
              icon="el-icon-edit"
              @click="editDetails()">
              编辑
            </el-button>
          </el-col>
        </el-row>
      </el-card>
    </el-main>
  </el-container>
</template>

<script>
export default {
  name: 'ShowActivityInformationPage',
  data () {
    return {
      currentActivity: [],
      width: 7,
      offset: 2,
      page_size: 20,
      currentPage: 1,
      clazz: [],
      clazzOptions: []
    }
  },
  created () {
    console.log(this.clazz)
    this.currentActivity = this.$store.getters.getCurrentActivity
    if (this.axios) {
      // this.findAllClasses()
      this.findExistClasses()
    }
  },
  methods: {
    goBack  () {
      this.$router.go(-1)
    },
    editDetails () {
      this.$router.push('/edit-activity')
    },
    findExistClasses () {
      let that = this
      let params = {
        page: that.currentPage,
        page_size: that.page_size,
        activity_id: that.currentActivity.id
      }
      that.axios.get('/activity/clazz', { params }).then((response) => {
        if (response.data.code === '2000') {
          let activityResults = response.data.data.results
          for (let i = 0; i < activityResults.length; i++) {
            that.clazz.push(activityResults[i].id)
          }
        }
        that.$forceUpdate()
      })
    }
  }
}
</script>

<style scoped>
  .el-container {
    font-size: 20px;
  }

  .description {
    font-size: 21px;
    text-align: center;
    color: black;
  }

  .center-position {
    text-align: center;
  }

  .line {
    margin: 0 10%;
  }

  .block {
    margin: 30px auto 0;
  }

  .el-col > i {
    font-size: 14px;
    color: #272324;
  }

  .el-col > label {
    margin-bottom: 100px;
    font-size: 14px;
  }

  .info-card {
    width: 62%;
    margin: auto;
  }

  .info-table {
    margin: 40px 20px;
  }

  .top-distance {
    margin-top: 30px;
  }

  .bottom-distance {
    margin-bottom: 40px;
  }
</style>
