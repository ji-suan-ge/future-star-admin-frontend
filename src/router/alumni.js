import ShowAlumniInformationPage from '../page/alumni-management/ShowAlumniInformationPage'
import EditAlumniInformationPage from '../page/alumni-management/EditAlumniInformationPage'
import ShowAlumniEditHistoryPage from '../page/alumni-management/ShowAlumniEditHistoryPage'
import ShowAlumniHistoryInformationPage from '../page/alumni-management/ShowAlumniHistoryInformationPage'

export default [
  {
    path: '/show-all-information',
    name: 'ShowAlumniInformationPage',
    component: ShowAlumniInformationPage
  },
  {
    path: '/edit-alumni-information',
    name: 'EditAlumniInformationPage',
    component: EditAlumniInformationPage
  },
  {
    path: '/show-edit-history',
    name: 'ShowAlumniEditHistoryPage',
    component: ShowAlumniEditHistoryPage
  },
  {
    path: '/show-history-information',
    name: 'ShowAlumniHistoryInformationPage',
    component: ShowAlumniHistoryInformationPage
  }
]
