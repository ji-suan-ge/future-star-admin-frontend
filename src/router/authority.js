import AddAdministratorPage from '../page/authority-management/AddAdministratorPage'

export default [
  {
    path: '/add-administrator',
    name: 'AddAdministratorPage',
    component: AddAdministratorPage
  }
]
