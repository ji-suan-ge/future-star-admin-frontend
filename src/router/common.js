import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import homeChildren from './home-children'

export default [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      ...homeChildren
    ]
  }
]
