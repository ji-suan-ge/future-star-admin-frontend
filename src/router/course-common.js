import DetailOfSemesterPage from '../page/course-common/DetailOfSemesterPage'
import DetailOfClassPage from '../page/course-common/DetailOfClassPage'
import DetailOfCoursePage from '../page/course-common/DetailOfCoursePage'
import DetailOfTeacherPage from '../page/course-common/DetailOfTeacherPage'
import DetailOfSubCoursePage from '../page/course-common/DetailOfSubCoursePage'
import EditSemesterInfoPage from '../page/course-common/EditSemesterInfoPage'
import EditClassInfoPage from '../page/course-common/EditClassInfoPage'
import EditCourseInfoPage from '../page/course-common/EditCourseInfoPage'
import EditTeacherInfoPage from '../page/course-common/EditTeacherInfoPage'
import AddCoursePage from '../page/course-common/AddCoursePage'
import AddCourseResourcePage from '../page/course-common/AddCourseResourcePage'
import StudentInClass from '../page/course-common/StudentInClass'

export default [
  {
    path: '/semester-detail',
    name: 'DetailOfSemesterPage',
    component: DetailOfSemesterPage
  },
  {
    path: '/class-detail',
    name: 'DetailOfClassPage',
    component: DetailOfClassPage
  },
  {
    path: '/course-detail',
    name: 'DetailOfCoursePage',
    component: DetailOfCoursePage
  },
  {
    path: '/teacher-detail',
    name: 'DetailOfTeacherPage',
    component: DetailOfTeacherPage
  },
  {
    path: '/sub-course-detail',
    name: 'DetailOfSubCoursePage',
    component: DetailOfSubCoursePage
  },
  {
    path: '/edit-semester-info',
    name: 'EditSemesterInfoPage',
    component: EditSemesterInfoPage
  },
  {
    path: '/edit-class-info',
    name: 'EditClassInfoPage',
    component: EditClassInfoPage
  },
  {
    path: '/edit-course-info',
    name: 'EditCourseInfoPage',
    component: EditCourseInfoPage
  },
  {
    path: '/edit-teacher-info',
    name: 'EditTeacherInfoPage',
    component: EditTeacherInfoPage
  },
  {
    path: '/add-course',
    name: 'AddCoursePage',
    component: AddCoursePage
  },
  {
    path: '/add-course-resource',
    name: 'AddCourseResourcePage',
    component: AddCourseResourcePage
  },
  {
    path: '/student-in-class',
    name: 'StudentInClass',
    component: StudentInClass
  }
]
