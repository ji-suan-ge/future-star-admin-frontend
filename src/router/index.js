import Vue from 'vue'
import VueRouter from 'vue-router'
import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import EnrollmentManagementPage from '../page/EnrollmentManagementPage'
import CourseManagementPage from '../page/CourseManagementPage'
import ActivityManagementPage from '../page/ActivityManagementPage'
import AlumniManagementPage from '../page/AlumniManagementPage'
import AuthorityManagementPage from '../page/AuthorityManagementPage'
import CreateActivityPage from '../page/CreateActivityPage'
import AddAdministrator from '../page/AddAdministrator'
import WelcomePage from '../page/WelcomPage'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      {
        path: 'welcome',
        name: 'WelcomePage',
        component: WelcomePage
      },
      {
        path: 'enrollment-management',
        name: 'EnrollmentManagementPage',
        component: EnrollmentManagementPage
      },
      {
        path: 'course-management',
        name: 'CourseManagementPage',
        component: CourseManagementPage
      },
      {
        path: 'activity-management',
        name: 'ActivityManagementPage',
        component: ActivityManagementPage
      },
      {
        path: 'alumni-management',
        name: 'AlumniManagementPage',
        component: AlumniManagementPage
      },
      {
        path: 'authority-management',
        name: 'AuthorityManagementPage',
        component: AuthorityManagementPage
      },
      {
        path: 'add-administrator',
        name: 'AddAdministrator',
        component: AddAdministrator
      }
    ]
  },
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  }
]

const router = new VueRouter({
  routes
})

export default router
