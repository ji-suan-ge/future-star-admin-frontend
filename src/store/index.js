import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    activeIndexOfNavigation: '0',
    activeNameOfCourse: 'first',
    activeNameOfEnrollment: 'first',
    activeNameOfActivity: 'first',
    activeSemesterOfCourse: '',
    semesters: '',
    currentSemester: '',
    classes: '',
    currentClass: '',
    courses: '',
    currentCourse: '',
    infoOfAnAlumni: null
  },
  mutations: {
    changeActiveIndexOfNavigation (state, activeIndex) {
      state.activeIndexOfNavigation = activeIndex
    },
    changeActiveNameOfEnrollment (state, activeName) {
      state.activeNameOfEnrollment = activeName
    },
    changeActiveNameOfCourse (state, activeName) {
      state.activeNameOfCourse = activeName
    },
    changeActiveNameOfActivity (state, activeName) {
      state.activeNameOfActivity = activeName
    },
    changeActiveSemesterOfCourse (state, activeID) {
      state.activeSemesterOfCourse = activeID
    },
    changeSemesters (state, semesters) {
      state.semesters = semesters
    },
    changeInfoOfAnAlumni (state, alumniInfo) {
      state.infoOfAnAlumni = alumniInfo
    },
    changeCurrentSemester (state, semester) {
      state.currentSemester = semester
    },
    changeClasses (state, classes) {
      state.classes = classes
    },
    changeCurrentClass (state, currentClass) {
      state.currentClass = currentClass
    },
    changeCourses (state, courses) {
      state.courses = courses
    },
    changeCurrentCourse (state, currentCourse) {
      state.currentCourse = currentCourse
    },
    initializeStateWhileLogin (state) {
      state.activeIndexOfNavigation = '0'
      state.activeNameOfCourse = 'first'
      state.activeNameOfEnrollment = 'first'
      state.activeNameOfActivity = 'first'
      state.activeSemesterOfCourse = ''
      state.semesters = ''
      state.currentSemester = ''
      state.classes = ''
      state.currentClass = ''
      state.courses = ''
      state.currentCourse = ''
      state.infoOfAnAlumni = null
    },
    initializeStateWhileSwitch (state) {
      state.activeNameOfCourse = 'first'
      state.activeNameOfEnrollment = 'first'
      state.activeNameOfActivity = 'first'
      state.activeSemesterOfCourse = ''
      state.semesters = ''
      state.currentSemester = ''
      state.classes = ''
      state.currentClass = ''
      state.courses = ''
      state.currentCourse = ''
      state.infoOfAnAlumni = null
    }
  },
  actions: {
    changeActiveIndexOfNavigation (context, activeIndex) {
      context.commit('changeActiveIndexOfNavigation', activeIndex)
    },
    changeActiveNameOfEnrollment (context, activeName) {
      context.commit('changeActiveNameOfEnrollment', activeName)
    },
    changeActiveNameOfCourse (context, activeName) {
      context.commit('changeActiveNameOfCourse', activeName)
    },
    changeActiveNameOfActivity (context, activeName) {
      context.commit('changeActiveNameOfActivity', activeName)
    },
    changeActiveSemesterOfCourse (context, activeID) {
      context.commit('changeActiveSemesterOfCourse', activeID)
    },
    changeSemesters (context, semesters) {
      context.commit('changeSemesters', semesters)
    },
    changeInfoOfAnAlumni (context, alumniInfo) {
      context.commit('changeInfoOfAnAlumni', alumniInfo)
    },
    changeCurrentSemester (context, semester) {
      context.commit('changeCurrentSemester', semester)
    },
    changeClasses (context, classes) {
      context.commit('changeClasses', classes)
    },
    changeCurrentClass (context, currentClass) {
      context.commit('changeCurrentClass', currentClass)
    },
    changeCourses (context, courses) {
      context.commit('changeCourses', courses)
    },
    changeCurrentCourse (context, currentCourse) {
      context.commit('changeCurrentCourse', currentCourse)
    },
    initializeStateWhileLogin (context) {
      context.commit('initializeStateWhileLogin')
    },
    initializeStateWhileSwitch (context) {
      context.commit('initializeStateWhileSwitch')
    }
  },
  getters: {
    getActiveIndexOfNavigation (state) {
      return state.activeIndexOfNavigation
    },
    getActiveNameOfEnrollment (state) {
      return state.activeNameOfEnrollment
    },
    getActiveNameOfCourse (state) {
      return state.activeNameOfCourse
    },
    getActiveNameOfActivity (state) {
      return state.activeNameOfActivity
    },
    getActiveSemesterOfCourse (state) {
      return state.activeSemesterOfCourse
    },
    getSemesters (state) {
      return state.semesters
    },
    getInfoOfAnAlumni (state) {
      return state.infoOfAnAlumni
    },
    getCurrentSemester (state) {
      return state.currentSemester
    },
    getClasses (state) {
      return state.classes
    },
    getCurrentClass (state) {
      return state.currentClass
    },
    getCourses (state) {
      return state.courses
    },
    getCurrentCourse (state) {
      return state.currentCourse
    }
  },
  modules: {
  }
})
