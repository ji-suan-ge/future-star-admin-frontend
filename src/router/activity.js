import CreateActivityPage from '../page/CreateActivityPage'
import ShowActivityParticipantsPage from '../page/ShowActivityParticipantsPage'
import EditExistingActivityPage from '../page/EditExistingActivityPage'
import ShowActivityInformationPage from '../page/ShowActivityInformationPage'

export default [
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  },
  {
    path: '/show-participants',
    name: 'ShowActivityParticipantsPage',
    component: ShowActivityParticipantsPage
  },
  {
    path: '/edit-activity',
    name: 'EditExistingActivityPage',
    component: EditExistingActivityPage
  },
  {
    path: '/show-activity-details',
    name: 'ShowActivityInformationPage',
    component: ShowActivityInformationPage
  }
]
