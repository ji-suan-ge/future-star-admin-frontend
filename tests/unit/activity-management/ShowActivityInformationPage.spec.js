import { createLocalVue, mount } from '@vue/test-utils'
import ShowActivityInformationPage from '../../../src/page/activity-management/ShowActivityInformationPage'
import Vuex from 'vuex'
import ElementUI from 'element-ui'
import VueRouter from 'vue-router'
import store from '../../../src/store'
import router from '../../../src/router'

describe('ShowActivityInformationPage', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.use(ElementUI)
  localVue.use(VueRouter)
  const wrapper = mount(ShowActivityInformationPage, {
    store,
    router,
    localVue
  })
  test('快照测试', async () => {
    await expect(wrapper.html()).toMatchSnapshot()
  })
  test('是一个 Vue 实例', async () => {
    await expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
