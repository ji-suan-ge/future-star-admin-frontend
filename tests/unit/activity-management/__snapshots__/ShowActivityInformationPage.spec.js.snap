// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowActivityInformationPage 快照测试 1`] = `
<section class="el-container">
  <main class="el-main">
    <div>
      <div class="el-page-header">
        <div class="el-page-header__left"><i class="el-icon-back"></i>
          <div class="el-page-header__title">返回</div>
        </div>
        <div class="el-page-header__content">活动详情</div>
      </div>
      <div class="el-divider el-divider--horizontal">
        <!---->
      </div>
    </div>
    <div class="el-card info-card is-never-shadow">
      <!---->
      <div class="el-card__body">
        <form class="el-form info-table">
          <div>
            <div class="el-divider el-divider--horizontal">
              <!---->
            </div>
          </div>
          <div class="description">- 活动信息 -</div>
          <div class="top-distance el-row">
            <div class="el-col el-col-7 el-col-offset-2"><i class="el-icon-present">&nbsp;活动名称：</i> <label></label></div>
            <div class="el-col el-col-7"><i class="el-icon-alarm-clock">&nbsp;报名开始时间：</i> <label></label></div>
            <div class="el-col el-col-7"><i class="el-icon-alarm-clock">&nbsp;报名结束时间：</i> <label></label></div>
          </div>
          <div class="top-distance el-row">
            <div class="el-col el-col-7 el-col-offset-2"><i class="el-icon-timer">&nbsp;活动时间：</i> <label></label></div>
            <div class="el-col el-col-7"><i class="el-icon-unlock">&nbsp;面向人群：</i> </div>
            <div class="el-col el-col-7"><i class="el-icon-map-location">&nbsp;活动地点：</i> <label></label></div>
          </div>
          <div class="top-distance el-row">
            <div class="el-col el-col-7 el-col-offset-2"><i class="el-icon-location-information">&nbsp;限制人数：</i> <label></label></div>
            <div class="el-col el-col-7"><i class="el-icon-coin">&nbsp;收费情况：</i> <label></label></div>
            <div class="el-col el-col-7"><i class="el-icon-mic">&nbsp;主办方：</i> <label></label></div>
          </div>
          <div class="top-distance el-row">
            <div class="el-col el-col-22 el-col-offset-2"><i class="el-icon-reading">&nbsp;活动日程：</i> <label></label></div>
          </div>
          <div class="top-distance el-row is-align-center">
            <div class="el-col el-col-24" align="center"><img width="400px;"></div>
          </div>
        </form>
        <div class="el-row">
          <div class="center-position el-col el-col-24"><button type="button" class="el-button el-button--primary is-round">
              <!----><i class="el-icon-edit"></i><span>
            编辑
          </span></button></div>
        </div>
      </div>
    </div>
  </main>
</section>
`;
